File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Sin.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class Sin extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "sin";
	}
	
	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.sin(input.doubleValue()));
	}
}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Sin.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal

class Sin extends IFunctionPlugin {

  def getName : String = "sin"
    
  def function(input: BigDecimal) : BigDecimal = {
  	val result:BigDecimal = new BigDecimal(Math.sin(input.doubleValue))
    result
  }
}